python_fundamentals-master/08_file_io/08_01_words_analysis.py

```diff
@@ -7,3 +7,20 @@
 
 
 '''
+with open("words.txt" , "r") as f:
+    word_min = "hello"
+    word_max = ""
+    words_min = []
+    words_max = []
+    total_num_words = 0
+    #not great, as it will not give the tie for word_max:
+    for word in f:
+        if len(word) <= len(word_min):
+            word_min = word
+            words_min.append(word_min.strip())
+        if len(word) >= len(word_max):
+            word_max = word
+        total_num_words += 1
+    print(words_min)
+    print(word_max)
+    print(total_num_words)
```

Comment:
So this doesn't actually meet the objective right? Anytime a word is larger or smaller it will get added to the right list. but the list will contain _all_ the words which were found in the process.  You need to find the _smallest_ of those words.  This could be the last one added (but then how how do you check for ties?) 
I think this needs a bit more work.